src/applications/disability-benefits/526EZ/helpers.jsx

```diff
@@ -272,6 +272,124 @@ const listifyCenters = (center, idx, list) => {
   );
 };
 
+export const documentTypeEnums = [
+  '1',
```

Comment:
@egarreau it looks like this is how `attachmentId`'s are enumerated in other forms, please let me know if we should approach this differently.

Reply:
It looks like we'll want to use the `L049`-type codes that the backend is already set up for here to simplify the transformation on the backend (so it will simply involve changing attachmentId to attachmentType, plus the existing hashing)?

Reply:
Currently, we've got duplicate `documentTypes` with different ids. This list on the back end doesn't have duplicates.

What I'd like to see is something more like what the back end's doing:
```js
export const documentTypes = {
  'L029': 'Copy of a DD214',
  'L450': 'STR - Dental - Photocopy',
  'L451': 'STR - Medical - Photocopy',
  'L049': 'Medical Treatment Record - Non-Government Facility',
  'L034': 'Military Personnel Record',
  'L107': 'VA Form 21-4142 - Authorization To Disclose Information',
  'L827': 'VA Form 21-4142a - General Release for Medical Provider Information',
  'L229': 'VA Form 21-0781a - Statement in Support of Claim for PTSD Secondary to Personal Assault',
  'L228': 'VA Form 21-0781 - Statement in Support of Claim for PTSD',
  'L149': 'VA Form 21-8940 - Veterans Application for Increased Compensation Based on Un-employability',
  'L115': 'VA Form 21-4192 - Request for Employment Information in Connection with Claim for Disability',
  'L159': 'VA Form 26-4555 - Application in Acquiring Specially Adapted Housing or Special Home Adaptation Grant',
  'L117': 'VA Form 21-4502 - Application for Automobile or Other Conveyance and Adaptive Equipment Under 38 U.S.C. 3901-3904',
  'L139': 'VA Form 21-686c - Declaration of Status of Dependents',
  'L133': 'VA Form 21-674 - Request for Approval of School Attendance',
  'L102': 'VA Form 21-2680 - Examination for Housebound Status or Permanent Need for Regular Aid & Attendance',
  'L222': 'VA Form 21-0779 - Request for Nursing Home Information in Connection with Claim for Aid & Attendance',
  'L702': 'Disability Benefits Questionnaire (DBQ)',
  'L703': 'Goldmann Perimetry Chart/Field Of Vision Chart',
  'L070': 'Photographs',
  'L023': 'Other Correspondence'
}
```
Then, if we need just the ids, we can do `Object.keys(documentTypes)` and when we want just the values, we can do `Object.values(documentTypes)`.

For the groups, since we no longer have duplicates, we can do something like this:
```js
export const documentTypeGroups = {
  'Automobile': ['L117'],
  'Individual Unemploy': [...],
  'Specially Adapted': [...],
  'Spouse Aid Attend': [...],
  'Special Monthly Comp': [...],
  'Temporary Total Disability - PTSD': [...],
  'Other': [...],
  'Claim Dependents': [...],
  'DBQ': [...],
  'DBQ Scar': [...],
  'DBQ Vision': [...]
}
```
Then what do we do with options that aren't in a group? :thinking: I think I'd like to put them at the top of the list, under no group that way it doesn't _look_ like they're under a group, y'know? That's where it might get a little tricky. To find those options, we should be able to do some fancy footwork like:
```js
/**
 * Loop through all options in groups to find options that don't have a group.
 */
function getUngroupedOptions(options, groups) {
  const ungroupedOptions = options.slice();
  Object.keys(groups).forEach(groupName => groups[groupName].forEach(id => {
    const index = ungroupedOptions.indexOf(id);
    if (index !== -1) {
      ungroupedOptions.splice(index, 1);
    }
  }));
  return ungroupedOptions;
}
```

Reply:
Sorry @cvalarida that comment is actually out dated, we will be sticking with the 1,2,3 enums since there are multiple display names/labels for some document types. Any additional feedback based on that is welcome however, looping in @egarreau as well to confirm.

Reply:
I mean using the document types rather than the 1,2,3 enums would be ideal for the backend but since it's not a one-to-one relationship between doc types and display names/labels I understand that's a challenge. If that's not an option, the 1,2,3 enums is a suitable second choice.

Reply:
Yeah, there would need to be a transformation either way, it sounds like the backend may be better suited for that?

Reply:
I like the general approach @markgreenburg outlines here, but because a single item may be in multiple groups, it might make sense to use `categories: ['PTSD', 'Something else']`. This would mean we couldn't use `_.groupBy()`, but building the list this way would make a lot of sense, and I think that's the part that matters when we're finished.

Provided we have good documentation and can reason about how it all works out when we look at the widget later, of course.